src/components/UserConfigurableDataGrid/useModelsFromQueryString.ts

```diff
@@ -11,9 +11,11 @@ import { useEffect, useState } from 'react';
 
 interface UseModelsFromQueryString {
   filterModel: GridFilterModel;
-  gridProps: Pick<
-    DataGridProProps,
-    'filterModel' | 'onFilterModelChange' | 'onSortModelChange' | 'sortModel'
+  gridProps: Required<
+    Pick<
+      DataGridProProps,
+      'filterModel' | 'onFilterModelChange' | 'onSortModelChange' | 'sortModel'
+    >
```

Comment:
This is one way to do it for sure, but I think I would have just done this:

```ts
interface UseModelsFromQueryString {
  filterModel: GridFilterModel;
  gridProps: {
    filterModel: DataGridPropProps['filterModel'],
    onFilterModelChange: DataGridProProps['onFilterModelChange'],
    onSortModelChange: DataGridProProps['onSortModelChange'],
    sortModel: DataGridProProps['sortModel'],
  },
  setFilterModel: (model: GridFilterModel) => void;
  setSortModel: (model: GridSortModel) => void;
  sortModel: GridSortModel;
}
```

Honestly having seen both, I'm not even sure which one I prefer. Just putting this comment here to share my thoughts, no need to change!

Reply:
hmmm.... I'm not so familiar with the style you present here, bc I'm not so advanced in TypeScript, but as I intrepret it it means f.x. "The type of filterModel in gridProps, is the same as the type that is assigned to the key 'filterModel' in DataGridProProps"
In that sense, the difference is that where it is optional in DataGridPropProps, I made them required. I could not see that the hook could ever return an object without a "full set" of properties, and it simplified the code elsewhere.
So if I have interpreted correctly what your suggestion does, I think mine is more clear because it communicates that there is a difference and the difference is that here these properties are required.

Reply:
oh nooow i see ok, ofcourse they are required in yours as well! ok, well, then they are just two versions of the same. I like mine because I'm proud of figuring out how to combine typescript tricks, but I am not sure which one is more easily read-able.